src/Umbraco.Web.BackOffice/DependencyInjection/UmbracoBuilder.LocalizedText.cs

```diff
@@ -77,14 +73,43 @@ private static IUmbracoBuilder AddSupplemenataryLocalizedTextFileSources(this IU
         {
             // locate all the *.xml files inside Lang folders inside folders of the main folder
             // e.g. /app_plugins/plugin-name/lang/*.xml
+            var fileSources = new List<LocalizedTextServiceSupplementaryFileSource>();
+
+            var pluginFolders = fileProvider.GetDirectoryContents(folder)
+                    .Where(x => x.IsDirectory).ToList();
+
+            foreach (var pluginFolder in pluginFolders)
+            {
+                // get the full virtual path for the plugin folder
+                var pluginFolderPath = WebPath.Combine(folder, pluginFolder.Name);
+
+                // get any lang folders in this plugin
+                var langFolders = fileProvider.GetDirectoryContents(pluginFolderPath)
+                    .Where(x => x.IsDirectory && x.Name.InvariantEquals("lang"));
+
+                // loop through the lang folder(s)
+                //  - there could be multiple on case sensitive file system
+                foreach (var langFolder in langFolders)
+                {
+                    // get the full 'virtual' path of the lang folder
+                    var langFolderPath = WebPath.Combine(pluginFolderPath, langFolder.Name);
+
+                    // request all the files out of the path, these will have physicalPath set.
+                    var files = fileProvider.GetDirectoryContents(langFolderPath)
+                        .Where(x => x.Name.InvariantEndsWith(".xml"))
+                        .Select(x => new FileInfo(x.PhysicalPath))
```

Comment:
This might cause `NullReferenceException`s, because the `IFileProvider`s abstraction of `IFileInfo` might not always have the [`PhysicalPath`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.fileproviders.ifileinfo.physicalpath) set (e.g. when it's retrieved form the `EmbeddedFileProvider`, `ManifestEmbeddedFileProvider` or external storage like Azure Blob Storage).

We basically shouldn't use `System.IO.FileInfo` anymore for read-only file access 😄

Reply:
We should at least make sure we don't try to create `FileInfo`s when the `PhysicalPath` is null or empty:
```suggestion
                        .Where(x => !string.IsNullOrEmpty(x.PhysicalPath))
                        .Select(x => new FileInfo(x.PhysicalPath))
```

Reply:
Yes - we are skirting close with the complete rewrite of the localization stuff i think, the only reason that the `FileInfo` is created is because `LocalizedTextServiceSupplementaryFileSource` requires the concrete class - we already have a `IFileInfo` from the fileprovider but i am not sure of the work to refactor the file sources to work with that vs `FileInfo`